src/kudu/consensus/metadata.proto

```diff
@@ -110,6 +116,10 @@ message RaftPeerPB {
   // Replica's health report, as seen by the leader. This is a run-time
   // only field, it should not be persisted or read from the persistent storage.
   optional HealthReportPB health_report = 5;
+
+  // Denotes if the server is backed by a database. If not, it only has the
+  // logs.
+  optional bool backing_db_present = 6 [ default = true ];
```

Comment:
Ideally no, consensus should not have to know about this. This boils down to the question of what information do we keep in the plugin, what is the persistence story around that or do we keep all the information within the configuration metadata. In that case backing_db_present or witness can become like a OPAQUE thing which kuduraft never acts on.

Reply:
Yesterday Ritwik mentioned offline that the purpose of this flag is to indicate whether a node is a witness for the purposes of graceful leadership stepdown, the idea being that if we can avoid making a witness the leader then we should try to. In that context it seems reasonable to me (but personally I think "witness" would make a better name for this rather than "backing_db_present")

Reply:
But I thought, we settled on "FOLLOWERS" and an extra set of attributes. Are all LBUs witnesses then, even the ones not attached to master?

Reply:
Fair point about only voters being witnesses. I'm fine with going with the name you chose here.